### PR TITLE
Generate shareable urls

### PR DESCRIPTION
**Pull request by philpep**
This is a first attempt to generate shareable urls (i.e. copying url and pasting to a new tab should display the same page in the same state).

TODO (might be in separate PR):

- [x] What about possible conflicting param names (commits, branch, idx, x-axis, y-axis) ? EDIT: fixed by prefixing params with `p-`
- [x] Replace "idx" (index of params permutation) in regression / summarylist by flat params
- [ ] store actual zoom state (EDIT: might be done in dedicated PR)

### CODE REVIEW

**Comment by philpep** [comment]
I fixed the two ongoing issues:

- Fix possible param name conflict by prefixing benchmark params with `p-`
- Replace the `idx` by flat param selection for generating regressions graph links.

I think the PR is ready for review & merge. Thansk!

**Comment by pv** [comment]
The links generated from "Regressions" page don't seem to work for me.

**Comment by philpep** [comment]
Thanks for testing this. I was able to reproduce a similar error message with a small screen height, do you still have the issue with #596 ?

**Comment by pv** [comment]
That fixed the JS exception, but there's still no plot shown.

File: asv/www/graphdisplay.js [hunk 1]
```diff
@@ -385,9 +385,19 @@ $(document).ready(function() {
...
 
                 for (var k = 0; k < params.length; ++k) {
+                    var param_name = param_names[k]
+                    var param_values = params[k]
                     var item = [];
-                    for (var j = 0; j < params[k].length && (j+1)*count <= max_curves; ++j) {
-                        item.push(j);
+                    if (state_selection['p-'+param_name] !== undefined) {
+                        for (var j = 0; j < param_values.length; ++j) {
+                            if (state_selection['p-'+param_name].includes($.asv.convert_benchmark_param_value(param_values[j]))) {
```
Comment (pv) [inline asv/www/graphdisplay.js]: `$.asv.convert_benchmark_param_value` does not always return a string, so this construct fails. The function probably should be changed to always return a string.

Comment (pv) [reply asv/www/graphdisplay.js]: Maybe the parameters should not use `$.asv.convert_benchmark_param_value` at all, and instead just use the raw string value.

Comment (philpep) [reply asv/www/graphdisplay.js]: Good catch, this was broken with non string parameters. Drop the use `$.asv.convert_benchmark_param_value` fix it.
I pushed new version of the fist patch, tested against numpy and mercurial suite.

**Comment by pv** [comment]
I now notice the Atom feed URL generation also needs updates to match 
the new URL scheme.

The `idx` parameter probably should be removed from the location hash if 
present, because its presence interferes with the parameter selector UI.

**Pull request merged**